src/foam/u2/DateView.js

```diff
@@ -34,23 +34,44 @@ foam.CLASS({
     [ 'placeholder', 'yyyy-mm-dd' ]
   ],
 
+  listeners: [
+    {
+      name: 'onBlur',
+      isFramed: true,
+      code: function() {
+        if ( ! this.el() ) return;
+        this.el().value = this.dataToInput(this.data);
+      }
+    }
+  ],
+
   methods: [
     function initE() {
       this.SUPER();
       this.setAttribute('type', 'date');
       this.setAttribute('placeholder', 'yyyy/mm/dd');
+      this.on('blur', this.onBlur);
     },
 
     function link() {
       this.data$.relateTo(
           this.attrSlot(null, this.onKey ? 'input' : null),
-          function(date) {
-            return date ? date.toISOString().substring(0,10) : date;
-          },
-          function(value) {
-            return new Date(value);
-          }
+          d => this.dataToInput(d),
+          d => this.inputToData(d)
       );
+    },
+
+    function inputToData(input) {
+      return new Date(input);
+    },
+
+    function dataToInput(data) {
+      // Using our own formatter to keep the date in the format (yyyy-mm-dd) while maintaining the locale date
+      const year = data.getFullYear();
+      const month = data.getMonth() + 1 > 9 ? data.getMonth() + 1 : `0${data.getMonth() + 1}`;
```

Comment:
I just discovered js's padStart that'll make life a little easier here:
const month = (data.getMonth() + 1).toString().padStart(2, '0');
const day = data.getDate().toString().padStart(2, '0');